pkg/controllers/node/termination/controller.go

```diff
@@ -94,114 +95,243 @@ func (c *Controller) finalize(ctx context.Context, node *corev1.Node) (reconcile
+		stored := nodeClaim.DeepCopy()
+		if modified := nodeClaim.StatusConditions().SetUnknownWithReason(v1.ConditionTypeVolumesDetached, "AwaitingVolumeDetachment", "AwaitingVolumeDetachment"); modified {
+			if err := c.kubeClient.Status().Patch(ctx, nodeClaim, client.MergeFromWithOptions(stored, client.MergeFromWithOptimisticLock{})); err != nil {
+				if errors.IsConflict(err) {
+					return &reconcile.Result{Requeue: true}, nil
+				}
+				if errors.IsNotFound(err) {
+					return &reconcile.Result{}, c.associatedNodeClaimError(err)
+				}
+				return &reconcile.Result{}, err
 			}
-			return reconcile.Result{}, fmt.Errorf("ensuring instance termination, %w", err)
 		}
-		if !isInstanceTerminated {
-			return reconcile.Result{RequeueAfter: 5 * time.Second}, nil
+		return &reconcile.Result{RequeueAfter: 1 * time.Second}, nil
+	} else {
+		// There are volume attachments blocking instance termination remaining, but the nodeclaim's TGP has expired. In this
+		// case we should set the status condition to false (requeing if it wasn't already) and then fall through to instance
+		// termination.
+		stored := nodeClaim.DeepCopy()
+		if modified := nodeClaim.StatusConditions().SetFalse(v1.ConditionTypeVolumesDetached, "TerminationGracePeriodElapsed", "TerminationGracePeriodElapsed"); modified {
+			if err := c.kubeClient.Status().Patch(ctx, nodeClaim, client.MergeFromWithOptions(stored, client.MergeFromWithOptimisticLock{})); err != nil {
+				if errors.IsConflict(err) {
+					return &reconcile.Result{Requeue: true}, nil
+				}
+				if errors.IsNotFound(err) {
+					return &reconcile.Result{}, c.associatedNodeClaimError(err)
+				}
+				return &reconcile.Result{}, err
+			}
+			// We requeue after a patch operation since we want to ensure we read our own writes before any subsequent
+			// operations on the NodeClaim.
+			return &reconcile.Result{RequeueAfter: 1 * time.Second}, nil
 		}
 	}
+	return nil, nil
+}
+
+// awaitInstanceTermination will initiate instance termination and continue to requeue until the cloudprovider indicates
+// the instance is no longer found. Once gone, the node's finalizer will be removed, unblocking the NodeClaim lifecycle
+// controller.
+func (c *Controller) awaitInstanceTermination(
+	ctx context.Context,
+	nodeClaim *v1.NodeClaim,
+	node *corev1.Node,
+	_ *time.Time,
+) (*reconcile.Result, error) {
+	isInstanceTerminated, err := termination.EnsureTerminated(ctx, c.kubeClient, nodeClaim, c.cloudProvider)
+	if client.IgnoreNotFound(err) != nil {
+		// 409 - The nodeClaim exists, but its status has already been modified
+		if errors.IsConflict(err) {
+			return &reconcile.Result{Requeue: true}, nil
+		}
+		return &reconcile.Result{}, fmt.Errorf("ensuring instance termination, %w", err)
+	}
+	if !isInstanceTerminated {
+		return &reconcile.Result{RequeueAfter: 5 * time.Second}, nil
+	}
 	if err := c.removeFinalizer(ctx, node); err != nil {
```

Comment:
Can we remove the finalizer outside of this function? Feels weird that each thing is waiting for a pre-condition, but only this one actually removes the finalizer at the end